crates/runtime/src/datafusion/query.rs

```diff
@@ -290,20 +310,21 @@ fn attach_query_context_to_stream(
     let schema_copy = Arc::clone(&schema);
 
     let mut num_records = 0u64;
-    let mut ctx = ctx;
 
     let updated_stream = stream! {
         while let Some(batch_result) = stream.next().await {
 
             match &batch_result {
                 Ok(batch) => {
                     num_records += batch.num_rows() as u64;
+                    yield batch_result
                 }
                 Err(e) => {
-                    ctx = ctx.error(e.to_string());
+                    ctx.finish_with_error(e.to_string(), ErrorCode::DataReadError).await;
```

Comment:
Found out that finalization code below is not called if there is an error so we need to finish query here.

```
ctx
            .schema(schema_copy)
            .rows_produced(num_records)
            .finish()
            .await;
```